src/event.rs

```diff
@@ -201,6 +222,24 @@ impl<'a, N: Notify + 'a> input::ActionContext for ActionContext<'a, N> {
     }
 }
 
+#[cfg(any(target_os = "freebsd", target_os = "openbsd"))]
+fn get_bsd_process_cwd(pid: u32) -> Result<String, String> {
+    let cwd = Command::new("procstat")
+        .args(&["-f", &pid.to_string()])
+        .output()
+        .expect("Failed to get current working directory.")
+        .stdout;
+    let cwd = std::str::from_utf8(&cwd)
+        .expect("Failed to parse current working directory.")
+        .split("\n");
+    let cwd = cwd.filter(|line| line.find("cwd").is_some())
+        .next();
+    match cwd {
+        Some(c) => Ok(c.to_string()),
+        None => Err("Failed to get BSD shell current working_directory.".to_string())
+    }
+}
```

Comment:
We shouldn't spawn a subprocess to a CLI application and attempt to parse its output. Instead the solution should be to call whatever the application itself is calling to acquire the required information.

Calling C code from Rust should be pretty straight-forward, so if there's a library on BSD which allows to retrieve this information, the Rust bindings to that C library can be generated automatically.

Reply:
What do you mean by log entries? if you are referring to all the previous commits that are appearing here.  I don't know why they are here. I'm sorry but this activity of contributing to projects is new to me. If you could tell me how avoid this from happening I will be grateful.